tests/Integration/Behaviour/Features/Context/Domain/ProductFeatureContext.php

```diff
@@ -472,6 +564,12 @@ public function assertProductFields(string $productReference, TableNode $table)
 
         $this->assertTaxRulesGroup($data, $productForEditing);
         $this->assertPriceFields($data, $productForEditing->getPricesInformation());
+        $this->assertShippingInformation($data, $productForEditing->getShippingInformation());
+
+        // Assertions checking isset() can hide some errors if it doesn't find array key,
+        // to make sure all provided fields were checked we need to unset every asserted field
+        // and finally, if provided data is not empty, it means there are some unnasserted values left
+        Assert::assertEmpty($data, sprintf('Some provided fields haven\'t been asserted: %s', implode(',', $data)));
```

Comment:
Nice one
I'm only afraid of one thing, in case you need to perform two checks on one field (or use it for anything else) it will have been cleared from the `$data` This might lead to unexpected behaviour for someone who is not aware of this cleaning (like "why did my data disappear?")

But for now, as long as we don't have issues this is perfect for the job

Reply:
Yea, we'll figure something out when time comes :sweat_smile: